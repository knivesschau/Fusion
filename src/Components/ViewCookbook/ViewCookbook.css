/* mobile styles first, min 320px */
.Your_Cookbook {
    text-align: center;
    margin: 2em;
}

.Welcome_Message {
    font-size: 1rem;
    background-color: rgb(252, 233, 217, .9);
    border: 1px solid rgb(128, 128, 128);
    border-radius: 1em;
    padding: .10em;
    box-shadow: inset -3px -2px 5px 0 white;
    text-shadow: 3px 2px 2px white;
    margin-left: auto;
    margin-right: auto;
    margin-top: -.5em;
    margin-bottom: 1.5em;
}

ul {
    list-style-type: none;
    padding: 0;
}

#cookbook-home {
    color: white; 
    text-shadow: 5px 3px 5px black;
    font-size: 2rem;
}

/* tablet and desktop styles */
@media only screen and (min-width: 768px) {
    .Welcome_Message {
        font-size: 1.25rem;
        padding: .5em;
        margin-left: 3.5em;
        margin-right: 3.5em;
    }
    
    #cookbook-home {
        font-size: 3rem;
    }
}

@media only screen and (min-width: 1200px) {
    .Welcome_Message {
        font-size: 1.5rem;
        margin-left: 15em;
        margin-right: 15em;
        margin-top: -1em;
    }

    #cookbook-home {
        font-size: 4rem;
    }
}

@media only screen and (min-width: 1600px) {
    .Welcome_Message {
        margin-left: 25em;
        margin-right: 25em;
        margin-top: -1.5em;
    }
}