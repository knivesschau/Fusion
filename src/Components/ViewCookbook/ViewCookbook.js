import React, { Component } from 'react';
import fusionContext from '../../fusionContext';
import RecipeInfo from '../RecipeInfo/RecipeInfo';
import './ViewCookbook.css';

export default class ViewCookbook extends Component {
    static contextType = fusionContext; 

    render() {
        
        const {fusions=[]} = this.context; 

        return (
            <section className="Your_Cookbook">
                
                <h1 id="cookbook-home">Your Cookbook</h1>

                <div className="Welcome_Message">
                    <span id="welcome">Welcome! Look at your recipes below, or click "Fuse" to make one!</span>
                </div>

                <ul>
                    {/* pass through context to RecipeInfo Component via props */}
                    {fusions.map(fusion =>
                        <li key={fusion.fused_id}>
                            <RecipeInfo
                                fused_id={fusion.fused_id}
                                fused_name={fusion.fused_name}
                                date_created={fusion.date_created}
                                base_cuisine={fusion.base_cuisine}
                                fuse_cuisine={fusion.fuse_cuisine}
                            />
                        </li>
                    )}
                </ul>

            </section>
        );
    };
};