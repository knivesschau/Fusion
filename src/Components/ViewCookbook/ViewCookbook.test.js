import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import ViewCookbook from './ViewCookbook';

describe ('ViewCookbook Component', () => {
    it ('Renders the complete ViewCookbook Component and all nested Components', () => {
        const wrapper = shallow(<ViewCookbook/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});