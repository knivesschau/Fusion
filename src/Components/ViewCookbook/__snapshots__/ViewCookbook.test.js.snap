// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewCookbook Component Renders the complete ViewCookbook Component and all nested Components 1`] = `
<section
  className="Your_Cookbook"
>
  <h1
    id="cookbook-home"
  >
    Your Cookbook
  </h1>
  <div
    className="Welcome_Message"
  >
    <span
      id="welcome"
    >
      Welcome! Look at your recipes below, or click "Fuse" to make one!
    </span>
  </div>
  <ul />
</section>
`;
